File: Iteration-2/src/Lecturer.java

import java.util.ArrayList;
import java.util.List;
public class Lecturer extends Person{

    private List<Course> assignedCourses;

    public Lecturer(String firstName, String lastName, List<Course> assignedCourses) {
        super(firstName, lastName);
        this.assignedCourses = new ArrayList<Course>();
    }

    public List<Course> getAssignedCourses() {
        return assignedCourses;
    }

    public void setAssignedCourses(Course course) {
        this.assignedCourses.add(course);
    }
    public String createEmailAddress() {
        return getFirstName().toLowerCase() + getLastName().toLowerCase() + "@marmara.edu.tr" ;
    }
}

File: Iteration-3/Src/Lecturer.py
from Person import Person

class Lecturer(Person):
    def __init__(self, first_name, last_name, assigned_courses):
        super().__init__(first_name, last_name)
        self.assigned_courses = []

    def get_assigned_courses(self):
        return self.assigned_courses

    def set_assigned_courses(self, course):
        self.assigned_courses.append(course)

    def create_email_address(self):
        return self.first_name.lower() + self.last_name.lower() + "@marmara.edu.tr"
